contracts/InterchainTokenFactory.sol

```diff
@@ -181,14 +182,19 @@ contract InterchainTokenFactory is IInterchainTokenFactory, ITokenManagerType, M
         tokenSymbol = token.symbol();
         tokenDecimals = token.decimals();
 
-        if (minter != address(0)) {
-            if (!token.isMinter(minter)) revert NotMinter(minter);
-            if (minter == address(interchainTokenService)) revert InvalidMinter(minter);
+        if (localMinter != address(0)) {
```

Comment:
What is the point of localMinter? You do some checks with it and do nothing with it, so there's not reason to not pass `address(0)` to the function.

Reply:
had discussion, updated PR